src/conventional/amv_ssec_ascii2ioda.py

```diff
@@ -0,0 +1,320 @@
+    writer = iconv.IodaWriter(output_file, locationKeyList, DimDict)
+    writer.BuildIoda(ioda_data, varDims, varAttrs, GlobalAttrs)
+
+    logging.info("--- {:9.4f} total seconds ---".format(time.time() - start_time))
+
+
+def read_file(file_name, data):
+
+    with open(file_name, newline='') as f:
+        reader = csv.DictReader(f, skipinitialspace=True, delimiter=' ')
+        keyerr = False
+        freqerr = False
+        saterr = False
+
+        for row in reader:
+            try:
+                year = int(row['day'][0:4])
+                month = int(row['day'][5:6])
+                day = int(row['day'][7:8])
+                hour = int(row['hms'][0:1])
+                minute = int(row['hms'][2:3])
+                second = 0
+                dtg = datetime(year, month, day, hour, minute, second)
+                time_offset = np.int64(round((dtg - epoch).total_seconds()))
+                data['dateTime'] = np.append(data['dateTime'], time_offset)
+                data['longitude'] = np.append(data['longitude'], float(row['lon']))
+                data['latitude'] = np.append(data['latitude'], float(row['lat']))
+
+                freq, freqerr = get_frequency(row['type'], freqerr)
+                satid, saterr = get_id(row['sat'], saterr)
+                data['satelliteID'] = np.append(data['satelliteID'], satid)
+                data['sensorCentralFrequency'] = np.append(data['sensorCentralFrequency'], freq)
+                data['sensorZenithAngle'] = np.append(data['sensorZenithAngle'], float(row['rff']))
+                data['windTrackingCorrelation'] = np.append(data['windTrackingCorrelation'], float(row['qi']))
+                data['windHeightAssignMethod'] = np.append(data['windHeightAssignMethod'], int(row['int']))
+
+                pres = float(row['pre'])*100.
+                data['air_pressure'] = np.append(data['air_pressure'], pres)
+                wdir = float(row['dir'])*1.0
+                wspd = float(row['spd'])*1.0
+                if (wdir >= 0 and wdir <= 360 and wspd >= 0 and wspd < 300):
+                    uwnd, vwnd = meteo_utils.meteo_utils().dir_speed_2_uv(wdir, wspd)
+                else:
+                    uwnd = float_missing_value
+                    vwnd = float_missing_value
+
+                data['eastward_wind'] = np.append(data['eastward_wind'], uwnd)
+                data['northward_wind'] = np.append(data['northward_wind'], vwnd)
+
+            except KeyError as e:
+                if not keyerr:
+                    logging.warning(file_name + ' is missing variable ' + e.args[0])
+                keyerr = True
+
+                if (e.args[0] == 'dir') | (e.args[0] == 'spd'):
+                    data['eastward_wind'] = np.append(data['eastward_wind'], float_missing_value)
+                    data['northward_wind'] = np.append(data['northward_wind'], float_missing_value)
+                else:
+                    outname, missing = get_outname(e.args[0])
```

Comment:
the `get_outname` function returns uninitialized values if the e.args[0] string doesn't match what the function is testing for. Should a check be put in the function for a non-match and a flag or special `outname` be returned so you can avoid putting a bad entry in the `data` dictionary? Or is this getting checked somewhere else so when you are here, you know you have a match?

Reply:
I'm not sure I understand. Do you mean if the file has a variable that's not one of the known ones in the test file? I'm not sure what would be best to do in that situation. get_outname contains all the known variables besides spd and dir

Reply:
I am concerned that if you encounter a variable from the file that isn't in the list that get_outname checks, you might end up with some garbage in the output ioda file and more importantly the user won't be aware that an unknown file variable was encountered.

If it's just a matter of keeping things consistent within this script I think get_outname is okay as is. What I mean is that there is a list of variables that are pulled from the file during the read, and that list is in sync with the list that get_outname checks. If this is the case, I think it would be helpful to put in comments where the variables are being extracted from the file and in get_outname pointing out that these two sections of code need to remain in sync. If a new variable is added to the section reading from the file, then it has to also be added in get_outname for example.

However, if a new variable is introduced by simply adding it to the file, then I think get_outname (or a new check) needs to handle unknown variables (in a similar fashion as get_frequency and get_id) so that the user becomes aware that a new variable is not being handled properly.

Reply:
I understand now. I have added a dictionary of known variables and section to print out any unknown variables in the file. The except block is similar since it would only reach this section of code while trying to write out one of the known variables that wasn't present (ie in the event of a KeyError in the code above)